### PR TITLE
Add OpenFaas to kbrew registry

### LINKED ISSUE #37: Recipe: OpenFaas
Hi, 

i would like to add  OpenFaas https://artifacthub.io/packages/helm/openfaas/openfaas to kbrew.

### PR DESCRIPTION
**Pull request by dirien**
Fixes #37 

Hope it is right.

Kind regards,

How i tested it localy:

```bash
➜  kbrew-registry git:(main) ✗ cp recipes/openfaas.yaml /Users/dirien/.kbrew/registries/kbrew-dev/kbrew-registry/recipes/
➜  kbrew-registry git:(main) ✗ kbrew install openfaas                                                                    
🚀 Installing openfaas app...
Version: 8.0.2
Pre-install dependencies: 
Post-install dependencies: 
---
 ✓ Installing app openfaas in openfaas namespace 
➜  kbrew-registry git:(main) ✗ kbrew remove openfaas
🧹 Uninstalling openfaas app and its dependencies...
Dependencies: 
---
 ✓ Removing app openfaas from openfaas namespace 
```

### CODE REVIEW

File: recipes/openfaas.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  repository:
+    name: openfaas
+    url: https://openfaas.github.io/faas-netes/
+    type: helm
+  namespace: "openfaas"
+  sha256:
+  version: 8.0.2
+  pre_install:
+  - steps:
+    - kubectl create ns openfaas-fn
```
Comment (alexellis) [inline recipes/openfaas.yaml:16]: @dirien 
The namespace creation should be done via kubectl apply https://.... and using the namespaces from the faas-netes repo

File: recipes/openfaas.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: kbrew
+app:
+  args:
+    functionNamespace: openfaas-fn
```
Comment (alexellis) [inline recipes/openfaas.yaml:5]: How do you override these at runtime?

File: recipes/openfaas.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    generateBasicAuth: true
```
Comment (alexellis) [inline recipes/openfaas.yaml:6]: What other elements are important from values.yaml? imagePullPolicy for instance

File: recipes/openfaas.yaml (hunk 1 above)
Comment (dirien) [reply recipes/openfaas.yaml]: done

File: recipes/openfaas.yaml (hunk 3 above)
Comment (dirien) [reply recipes/openfaas.yaml:6]: I took the defaults from https://artifacthub.io/packages/helm/openfaas/openfaas.

Any suggestion?

Comment (alexellis) [reply recipes/openfaas.yaml:6]: I suggested one above the Helm chart README is the best place to look - https://github.com/openfaas/faas-netes/blob/master/chart/openfaas/README.md - or values.yaml

Comment (dirien) [reply recipes/openfaas.yaml:6]: most of them have already default, reasonable values. I always start openfaas like this for lokal dev. And it works perfectly.

**Comment by dirien** [comment]
@PrasadG193, can you have a look when you have time? Interessted if the recipe is okay so.

File: recipes/openfaas.yaml (hunk 2 above)
Comment (PrasadG193) [reply recipes/openfaas.yaml:5]: We are already working on this :) 
https://github.com/kbrew-dev/kbrew/issues/122

File: recipes/openfaas.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    type: helm
+  namespace: "openfaas"
+  sha256:
+  version: 8.0.2
+  pre_install:
+  - steps:
+    - kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
+  post_cleanup:
+    steps:
+    - kubectl delete ns openfaas-fn
```
Comment (PrasadG193) [inline recipes/openfaas.yaml:19]: I think we should delete `openfaas` namespace too?

Comment (dirien) [reply recipes/openfaas.yaml]: hi @PrasadG193 ,  i thought it gets delete autmaticaly, wenn the chart is gone! Ok np!

File: recipes/openfaas.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    - kubectl delete ns openfaas
```
Comment (PrasadG193) [inline recipes/openfaas.yaml:20]:
```suggestion
    - kubectl delete ns openfaas

```

**Review by PrasadG193 — APPROVED** [review]

**Comment by sahil-lakhwani** [comment]
@dirien Recipe LGTM. Is there a quick way to verify if the installation was correct? If there, it require some wait?

**Comment by dirien** [comment]
hi @sahil-lakhwani , the web ui should appear on http://<ip>:8080. Depending on how long it take to pull the image and startup time of the service.

**Comment by sahil-lakhwani** [comment]
> hi @sahil-lakhwani , the web ui should appear on http://:8080. Depending on how long it take to pull the image and startup time of the service.

Ok, then I think there's no definitive way without waiting. We can go without that for now.

**Review by sahil-lakhwani — APPROVED** [review]

**Comment by dirien** [comment]
hi @sahil-lakhwani: Small ping 😄

**Comment by sahil-lakhwani** [comment]
hey @dirien , I was out last till yesterday. Merging this. Thanks a lot!

**Comment by dirien** [comment]
@sahil-lakhwani great thanks. FYI @alexellis !

**Pull request merged**